### PR TITLE
Replace QuickContactBadge with custom widget

### PR DESCRIPTION
**Pull request by Valodim**
See #1580, thanks to @madrat-. I only cleaned up a bit, and put a note about the origin of the file instead of the copyright notice.

This is only a cosmetic change, however it prepares round contact pictures, and a gmail-style click-to-select feature.

### CODE REVIEW

**Comment by philipwhiuk** [comment]
The code looks fine to me. I'll try and review the actual visible changes this evening.

File: k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            } finally {
+                if (cursor != null) {
+                    cursor.close();
+                }
+            }
+
+            contactUri = lookupUri;
+            onContactUriChanged();
+
+            if (trigger && lookupUri != null) {
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java]: could move this `if() else` block into a seperate function to keep this more less massive.

Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java]: I kept the code structure the same (other than removing stuff), to keep it aligned and mergable with the original QuickContactBadge source.

File: k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    @Override
+    protected void drawableStateChanged() {
+        super.drawableStateChanged();
+    }
+
+    /**
+     * This call has no effect anymore, as there is only one QuickContact mode
+     */
+    @SuppressWarnings("unused")
+    public void setMode(int size) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java]: This method doesn't seem to be used from any of our code. And it doesn't override a method of the parent class. Why is it here?

File: k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    }
+
+    public ContactBadge(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        queryHandler = new QueryHandler(context.getContentResolver());
+        setOnClickListener(this);
+    }
+
+    @Override
+    protected void drawableStateChanged() {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java]: This doesn't change the visibility of the method and only calls through to the super implementation. Can be removed.

File: k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+    /**
+     * This call has no effect anymore, as there is only one QuickContact mode
+     */
+    @SuppressWarnings("unused")
+    public void setMode(int size) {
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java]: This doesn't change the visibility of the method and only calls through to the super implementation. Can be removed.

File: k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+
+/**
+ * ContactBadge replaces the android ContactBadge for custom drawing.
+ * <p>
+ * Based on QuickContactBadge:
+ * https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/QuickContactBadge.java
+ */
+public class ContactBadge extends ImageView implements OnClickListener {
+    static final private int TOKEN_EMAIL_LOOKUP = 0;
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java]: Modifiers in wrong order.

File: k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        }
+    }
+
+    private void onContactUriChanged() {
+        setEnabled(isAssigned());
+    }
+
+    @Override
+    public void onClick(View v) {
+        // If contact has been assigned, mExtras should no longer be null, but do a null check
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/ui/ContactBadge.java]: Reference to "mExtras".

**Pull request merged**